src/lib/CollectStore.js

```diff
@@ -138,22 +138,27 @@ export default class CollectStore {
     return accounts.getAccountsByType(cozy.client, accountType, index)
   }
 
-  updateAccount (connectorId, accountIdx, values) {
-    let connector = this.findConnectorById(connectorId)
-    let previousAccount = Object.assign({}, connector.accounts[accountIdx])
-
+  /**
+   * updateAccount : updates an account in a connector in DB with new values
+   * @param {Object} connector The connector to update
+   * @param {integer} accountIdx The Index of the account to update in the accounts array of the connector
+   * @param {Object} values The new values of the updated account
+   * @returns {Object} The up to date connector
+   */
+  updateAccount (connector, accountIdx, values) {
```

Comment:
We should pass an account object instead of an index here, we cannot be sure that the array we will be looking at has not mutated since we get this id.